brainiak/utils/fmrisim.py

```diff
@@ -2744,6 +2800,22 @@ def generate_noise(dimensions,
                                               noise_dict=noise_dict,
                                               )
 
+    # Generate the drift noise
+    if noise_dict['drift_sigma'] != 0:
+        # Calculate the drift time course
+        noise = _generate_noise_temporal_drift(len(stimfunction_tr),
+                                               tr_duration,
+                                               )
+        # Create a volume with the drift properties
+        volume = np.ones(dimensions[:3])
+
+        # Combine the volume and noise
+        drift_noise = np.multiply.outer(volume, noise) * noise_dict[
```

Comment:
For consideration of future improvement: I think this line assumes that all voxels have the same time series of drift with the same magnitude. So I think it can capture what is removed by global signal regression. But I think in reality the drift may be of different sizes across the brain.

Reply:
Yeah completely agree, that is true. I thought about using a gaussian like with the other sources of noise but that also wouldn't be a good approximation. So yes would be a nice feature to add in the future